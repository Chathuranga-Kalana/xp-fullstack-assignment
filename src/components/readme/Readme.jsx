import React from "react";
import marked from "marked";
import "./readme.css";

import { github_raw_api } from "../../utils/api";

class Readme extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      readme: "",
      networkresp: true,
      name: this.props.match.params.username,
      reponame: this.props.match.params.reponame
    };
  }

  componentDidMount() {
    // alert(this.props.match.params.username);
    const { name, reponame } = this.state;

    // get readme raw data
    github_raw_api(`/${name}/${reponame}/master/README.md`)
      .then(response => {
        this.setState({ networkresp: response.ok });
        return response.text();
      })
      .then(text => {
        this.setState({ markdown: marked(text) });
      })
      .catch(error => {
        console.error(error);
        return { name: "Network Error" };
      });
  }
  render() {
    const { markdown } = this.state;
    const networkresp = this.state.networkresp;
    return (
      <section>
        <h3 className="readme_title"> Readme.md</h3>
        {/* check readme file is available */}
        {networkresp ? (
          <article dangerouslySetInnerHTML={{ __html: markdown }}></article>
        ) : (
          <p>File Not Found</p>
        )}
      </section>
    );
  }
}

export default Readme;
