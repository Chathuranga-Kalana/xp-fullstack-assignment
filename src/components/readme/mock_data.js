export const username = "Chathuranga-Kalana";
export const repo_name = "Nunchi";
export const mockReadme = `# Nunchi

Nunchi is machine learning absenteeism hour prediction web application .
This web applicaton shows most reasons for why employees getting absent by showing graphs individual or overall.
It also generate PDF files

## prerequisite
- Python 3
- flask
- weasyprint(https://weasyprint.readthedocs.io/en/latest/install.html#gtk64installer)
- pandas
- numpy
- matplotlib
- seaborn`;
