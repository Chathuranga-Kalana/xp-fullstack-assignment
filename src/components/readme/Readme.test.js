import React from "react";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { username, repo_name, mockReadme } from "./mock_data";
import Readme from "./Readme";

configure({ adapter: new Adapter() });

global.fetch = jest.fn().mockImplementation(path => {
  switch (path) {
    case `https://raw.githubusercontent.com/${username}/${repo_name}/master/README.md`:
      console.log(path);
      return Promise.resolve({
        ok: true,
        text: () => Promise.resolve({ mockReadme })
      });
    default:
      return {};
  }
});

describe("ReadmeComponent", () => {
  it("should fetch readme.md and render Readme component", async () => {
    const readme_comp = shallow(
      <Readme match={{ params: { username: username, reponame: repo_name } }} />
    );

    await new Promise(setImmediate);
    readme_comp.update();
    expect(readme_comp.getElements()).toMatchSnapshot();
  });
});
