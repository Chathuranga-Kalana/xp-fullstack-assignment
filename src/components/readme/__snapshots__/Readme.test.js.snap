// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadmeComponent should fetch readme.md and render Readme component 1`] = `
Array [
  <section>
    <h3
      className="readme_title"
    >
       Readme.md
    </h3>
    <article
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
    />
  </section>,
]
`;
